### PR TITLE
1.4 - State tracking for Phylum incomplete package data, fixed yarn parsing, cleanup

### PR DESCRIPTION
**Pull request by peterjmorgan**
This PR is ready for `v1.4` release

* Adds state tracking in a pull request to handle cases where Phylum returns incomplete risk analysis data allowing for resubmission by re-running the GitHub check (the instantiation of the action in a PR). When Phylum returns complete analysis, relevant comments are appended to the thread of the PR. 
* Adds test_matrix to invoke the testing for the above state tracking
* Improves yarn lockfile parsing. Now supports yarn v1 and v2 lockfiles and handles edge cases of in place package upgrades smoothly. Also begins the process of breaking out lockfile parsers into modules for unit tests.

### CODE REVIEW

File: analyze.py [hunk 1]
```diff
@@ -26,6 +21,35 @@
...
+    States on returncode
+    0 = No comment
+    1 = FAILED_COMMENT
+    5 = INCOMPLETE_COMMENT then:
+        4 = COMPLETE_SUCCESS_COMMENT
+        1 = COMPLETE_FAILED_COMMENT
+'''
+
+# Headers for distinct comment types
+DETAILS_DROPDOWN = "<details>\n<summary>Background</summary>\n<br />\nThis repository uses a GitHub Action to automatically analyze the risk of new dependencies added to requirements.txt via Pull Request. An administrator of this repository has set score requirements for Phylum's five risk domains.<br /><br />\nIf you see this comment, one or more dependencies added to the requirements.txt file in this Pull Request have failed Phylum's risk analysis.\n</details>\n\n"
```
Comment (louislang) [inline analyze.py:34]: Is this a message specific to Python? If it's not, we might want to change `requirements.txt` to whatever the common package lock name file is for each ecosystem.

File: analyze.py [hunk 2]
```diff
@@ -38,17 +62,21 @@ def __init__(self, repo, pr_num, vul, mal, eng, lic, aut):
...
     def get_env_vars(self):
         for key in ENV_KEYS:
             temp = os.environ.get(key)
             if temp is not None:
                 self.env[key] = temp
             else:
-                print(f"[ERROR] could not get value for os.environ.get({key})")
-                sys.exit(11)
+                print(f"[ERROR] could not get value for required env variable os.environ.get({key})")
+                # sys.exit(11) #TODO: re-enable
```
Comment (louislang) [inline analyze.py:77]: Caught this TODO, does this need to be re-enabled now or later?

File: analyze.py (hunk 1 above)
Comment (peterjmorgan) [reply analyze.py]: Fixed in https://github.com/phylum-dev/phylum-analyze-pr-action/commit/f6e17e90d6087c714c5672ce77ab9a236c097c21

**Review by furi0us333 — APPROVED** [review]
